cmake/checks/check_01_cpu_features.cmake

```
@@ -245,6 +246,37 @@ if(DEAL_II_ALLOW_PLATFORM_INTROSPECTION)
     "
     DEAL_II_HAVE_ALTIVEC)
 
+    CHECK_CXX_SOURCE_RUNS(
+      "
+      #include <arm_neon.h>
```

If we wanted to be really sure, we would add
```suggestion
      #ifndef __ARM_NEON
      #error Preprocessor flag not found
      #endif
      #include <arm_neon.h>
```